C-Plus-Plus/dp/BlockSumMatrix.cpp

```diff
@@ -0,0 +1,62 @@
+'''
+A matrix[m][n] & an integer k returns matrix where each matrix[i][j] is the sum of all element of matrix[r][c]
+for i - k <= r <= i+k , j - k <= c <= j+k where r,c are valid positions in matrix.
+'''
+
+// C++ Code:
+
+#include <bits/stdc++.h>
+using namespace std;
+void print_blocksum(vector<vector<auto> > sz)  // Block matrix representation of each sum
+{
+   cout << "[";
+   for(int i = 0; i<sz.size(); i++){
+      cout << "[";
+      for(int j = 0; j <sz[i].size(); j++){
+         cout << sz[i][j] << ", ";
+      }
+      cout << "],";
+   }
+   cout << "]"<<endl;
+}
+class Solution {
+public:
+   vector<vector<int>> matrixBlockSum(vector<vector<int>>& mat, int k) {
+      int n = mat.size();
+      int m = mat[0].size();
+      vector < vector <int> > res(n , vector <int> (m));  // Computing the matrix
+      for(int i = 0; i < n; i++){
+         for(int j = 0; j < m; j++){
+            for(int r = i - k;r <= i + k; r++){
+               for(int c = j - k; c <= j + k; c++){
+                  if(r>= 0 && r < n && c >= 0 && c < m){
+                     res[i][j] += mat[r][c]; //result add
+                  }
+               }
+            }
+         }
+      }
+      return res;
+   }
+};
+main(){
+   vector<vector<int>> sz1 = [[1,2,3], [4,5,6], [7,8,9]];  //put the input from test cases
```

Comment:
Take Dynamic input from the user with displaying  message in print form, as what are you taking input from the user